### PR TITLE
Move JMS connector code to Ballerina repo

### PR DESCRIPTION
**Pull request by a5anka**
## Purpose

Move JMS connector code to Ballerina repo from wso2-ballerina/package-jms.

### CODE REVIEW

File: stdlib/ballerina-jms/pom.xml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (keizer619) [inline stdlib/ballerina-jms/pom.xml]: @a5anka 2018

File: stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        preProcessIfWso2MB(configParams);
+        updateMappedParameters(configParams);
+
+        Properties properties = new Properties();
+        configParams.forEach(properties::put);
+
+        try {
+            InitialContext initialContext = new InitialContext(properties);
+            ConnectionFactory connectionFactory =
+                    (ConnectionFactory) initialContext.lookup(factoryName);
```
Comment (Shan1024) [inline stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java]: Can merge this line with above line.

File: stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+//    }
+
+    /**
+     * Process the provided properties in the {@link Map} and convert it to jms connector friendly Map.
+     *
+     * @param configParams Map instance that is getting filled.
+     * @param properties   {@link Map} of properties.
+     */
+    private static void preProcessMapField(Map<String, String> configParams,
+                                           Map<String, Value> properties) {
```
Comment (Shan1024) [inline stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java]: Can merge this line with above line.

File: stdlib/ballerina-jms/pom.xml (hunk 1 above)
Comment (a5anka) [reply stdlib/ballerina-jms/pom.xml]: This pom was originally created in 2017. Should I change?

File: stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java (hunk 2 above)
Comment (a5anka) [reply stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java]: :+1:

File: stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/nativeimpl/endpoint/session/InitEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package org.ballerinalang.net.jms.nativeimpl.endpoint.session;
```
Comment (keizer619) [inline stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/nativeimpl/endpoint/session/InitEndpoint.java]: @a5anka Include license headers

File: stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java (hunk 3 above)
Comment (a5anka) [reply stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java]: :+1:

File: stdlib/ballerina-jms/pom.xml (hunk 1 above)
Comment (Shan1024) [reply stdlib/ballerina-jms/pom.xml]: I don't think we have to change the year if the file was created in last year.

File: stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/nativeimpl/endpoint/session/InitEndpoint.java (hunk 4 above)
Comment (a5anka) [reply stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/nativeimpl/endpoint/session/InitEndpoint.java]: :+1:

File: stdlib/ballerina-jms/pom.xml (hunk 1 above)
Comment (a5anka) [reply stdlib/ballerina-jms/pom.xml]: :+1:

File: stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/nativeimpl/endpoint/session/InitEndpoint.java (hunk 4 above)
Comment (a5anka) [reply stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/nativeimpl/endpoint/session/InitEndpoint.java]: Fixed

**Pull request merged**